docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_testing-in-testing-farm.adoc

```diff
@@ -0,0 +1,51 @@
+= Adding an integration test
+
+link:https://docs.testing-farm.io/[Testing Farm] is a third-party test service that can be used as a test execution back-end for {ProductName}. This document guides you through an example of how to trigger Testing Farm tests from your {ProductName} *Integration tests*.
+
+.Prerequisites
+
+* You have created an application in {ProductName}.
+* You have a Testing Farm API key, which can be acquired by following the link:https://docs.testing-farm.io/Testing%20Farm/0.1/onboarding.html[Testing Farm onboarding guide].
+* You have have or are ready to create a git repository with link:https://fmf.readthedocs.io/[fmf] files suitable for use with the link:https://tmt.readthedocs.io/[tmt] testing tool. This is the testing technology used by Testing Farm under the hood. If you don't have a fmf tests already prepared, you can use our tmt link:https://github.com/ralphbean/tmt-hello-world[hello world repository].
+
+.Procedure
+
+You need to perform two major steps. Upload your Testing Farm API key, and register an *Integration test* to send requests to the Testing Farm API. Complete the following steps in the {ProductName} console:
+
+.Procedure - Upload your Testing Farm API key
+
+Follow similar instructions as in the xref:how-to-guides/configuring-builds/proc_creating-secrets-for-your-builds.adoc[Creating secrets for your builds] guide.
+
+. In {ProductName}, from the left navigation menu, select **Secrets**.
+. From the **Secrets** page, click **Add secret**.
+. From the **Add secret** page, choose **Build**.
+. Select secret type **Key/value secret**
+. For **Secret name**, enter `testing-farm-secret`.
+. Under **Key/value secret**, expand **Key/value 1**, then enter the key name `testing-farm-token`.
+NOTE: Be sure that the secret name is `testing-farm-secret` and the key name is `testing-farm-token`. The link:https://github.com/ralphbean/testing-farm-tekton/blob/main/tasks/testing-farm.yaml[tekton task] expects the secret and key name to be named these strings exactly.
+. For **Upload the file with value for your key or paste its contents**, paste the value of your token into the space under **Upload**.
+. Optional: Under **Labels**, add a label to tag or provide more context for your secret.
+. Click **Add secret**.
```

Comment:
Can we remove this duplication? Is the creating secret text insufficient to inform how to do this? If so, can we instead improve that text? There is already drift in configuring the secrets for the [Red Hat access token](https://redhat-appstudio.github.io/docs.appstudio.io/Documentation/main/how-to-guides/Import-code/proc_importing_code/#configuring-your-application-to-use-a-red-hat-container-registry-token).

If you duplicated so that you could ensure that the secret/token were named properly, then we could also have a parameter to specify the names of the secret/token which just defaults to `testing-farm-secret` and `testing-farm-token`. This would enable us to move that part of the documentation to specifying the parameters for the Integration Test Scenario.

Reply:
Yeah, I duplicated it 1) so that I could ensure that the secret/token was named properly and 2) so that the user didn't have to click to a second doc to complete the instructions.

Reply:
It would be great if we could inline other documentation segments so that they can be kept up to date but not be duplicated/require clicks...